docs: Update docs quickstart (#258)
block/goose: README.md

goose session start
```

#### Set up a provider

> Goose works on top of LLMs and by default, it uses `openai` as the LLM provider but you can customize it as needed. You'll be prompted to set an [OPENAI_API_KEY][openai-key] if you haven't set one previously.

Reviewer: ```suggestion
Goose works with your [preferred LLM](https://block.github.io/goose/plugins/providers.html). By default, it uses `openai` as the LLM provider. You'll be prompted to set an [API key][openai-key] if you haven't set one previously.
```

rephrasing a bit to let them know other LLMs are available. Don't want people to be DOA if they use something other than Open AI

>[!TIP]
> **Billing:**
>

> > You will need to add credits to your Open AI accounts to be able to successfully make requests.

Reviewer: ```suggestion
> You will need to have credits in your LLM Provider account to be able to successfully make requests.
```
